backend/src/routes/api/storage/index.ts

```diff
@@ -0,0 +1,65 @@
+import { FastifyInstance, FastifyReply } from 'fastify';
+import { createMinioClient, getObjectStream } from './storageUtils';
+import { getDashboardConfig } from '../../../utils/resourceUtils';
+import { getDirectCallOptions } from '../../../utils/directCallUtils';
+import { getAccessToken } from '../../../utils/directCallUtils';
+import { OauthFastifyRequest } from '../../../types';
+
+export default async (fastify: FastifyInstance): Promise<void> => {
+  fastify.get(
+    '/:namespace/:bucket',
+    async (
+      request: OauthFastifyRequest<{
+        Querystring: { key: string; peek?: number };
+        Params: { namespace: string; bucket: string };
+      }>,
+      reply: FastifyReply,
+    ) => {
+      const dashConfig = getDashboardConfig();
+      if (dashConfig?.spec.dashboardConfig.disableS3Endpoint === false) {
+        try {
+          const { namespace, bucket } = request.params;
+          const { key, peek } = request.query;
+
+          const requestOptions = await getDirectCallOptions(fastify, request, request.url);
+          const token = getAccessToken(requestOptions);
+
+          const stream = await getObjectStream({
```

Comment:
Can you get the stat information on the file before downloading it so we know how large it is... maybe make that another endpoint and allow this one as a text-stream endpoint.